internal/worker/externalcontrollerupdater/externalcontrollerupdater_test.go

```diff
@@ -181,6 +182,47 @@ func (s *ExternalControllerUpdaterSuite) TestWatchExternalControllersErrorRestar
 	workertest.CleanKill(c, w)
 }
 
+func (s *ExternalControllerUpdaterSuite) TestWatchExternalControllersNotSupported(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	ch := make(chan []string, 1)
+	extCtrlWatcher := watchertest.NewMockStringsWatcher(ch)
+	ch <- []string{coretesting.ControllerTag.Id()}
+
+	s.client.EXPECT().WatchExternalControllers().Return(extCtrlWatcher, nil)
+	info := &crossmodel.ControllerInfo{}
+	s.client.EXPECT().ExternalControllerInfo(coretesting.ControllerTag.Id()).Return(info, nil)
+
+	notSupportedErr := &params.Error{Code: params.CodeNotSupported}
+	watcherReady := make(chan struct{})
+	watcherFetched := make(chan struct{})
+
+	w, err := externalcontrollerupdater.New(s.client, func(*api.Info) (externalcontrollerupdater.ExternalControllerWatcherClientCloser, error) {
+		close(watcherReady)
+		<-watcherFetched
+		return nil, notSupportedErr
+	}, s.clock)
+	c.Assert(err, jc.ErrorIsNil)
+	defer workertest.CleanKill(c, w)
+
+	// Here we synchronise access to the controllerWatcher worker started
+	// by the runner in the updaterWorker. Fetch the single controllerWatcher
+	// worker from the the list of running workers before it is killed and
+	// removed, then check that it is killed with the expected error.
+	<-watcherReady
+	runner := externalcontrollerupdater.GetWorkerRunner(c, w)
+	names := runner.WorkerNames()
```

Comment:
You could add a public method to the external controller updater to get a named controller watcher and call that and remove the need for export test. Or just put this test in the package.